src/bundle.ts

```diff
@@ -0,0 +1,32 @@
+export default class Bundle {
+	public tarStream: NodeJS.ReadableStream
+
+	/**
+	 * deviceType: The slug of the device type that this bundle has been created
+	 * for
+	 */
+	public deviceType: string
+
+	/**
+	 * architecture: The architecture that this resin bundle is targeting
+	 */
+	public architecture: string
```

Comment:
@pimterry @CameronDiver @jviotti -- the work on contracts is producing a canonical set of architectures, languages, device types, etc. We should think of how we can use that in conjunction with TS perhaps?